crates/llm/src/lib.rs

```diff
@@ -187,3 +199,26 @@ pub fn load_dynamic(
 
     Ok(model)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_model_architecture_from_str() {
+        static ARCHS: [ModelArchitecture; 5] = [
```

Comment:
Maybe add this as an associated constant to `ModelArchitecture`, so that anyone can loop over `ModelArchitecture::ALL`?